### PR TITLE
Use `ArgumentBuilder` and `genCall` to cleanup codegen of `cp.async` and `ldmatrix`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
There should be no behavioral change, just make `codegen.cpp` more readable.

### CODE REVIEW

File: runtime/memory.cu [hunk 1]
```diff
@@ -77,44 +77,44 @@ DEVICE_INLINE void adjustPartialLdMatrixAddrInTuring(unsigned& addr_in_byte) {
 // warp.
 
 template <typename T>
-DEVICE_INLINE void ldMatrix(Array<T, 4, 4>& out, unsigned addr) {
+DEVICE_INLINE void ldMatrix(Array<T, 4, 4>* out, unsigned addr) {
```
Comment (naoyam) [inline runtime/memory.cu:80]: Does this make any impact?

**Comment by zasdfgbnm** [comment]
!build

File: runtime/memory.cu (hunk 1 above)
Comment (zasdfgbnm) [reply runtime/memory.cu:80]: I checked `ptxas_verbose`, the number of used registers are exactly the same. So I assume that if the existing SASS tests pass, then there shouldn't be anything wrong.

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**